src/SmartTransactionsController.ts

```diff
@@ -330,32 +396,48 @@ export default class SmartTransactionsController extends BaseController<
     });
   }
 
-  async updateSmartTransactions() {
+  async updateSmartTransactions({
+    networkClientId,
+  }: {
+    networkClientId?: NetworkClientId;
+  } = {}): Promise<void> {
     const { smartTransactions } = this.state.smartTransactionsState;
-    const { chainId } = this.config;
-
-    const currentSmartTransactions = smartTransactions?.[chainId];
+    const chainId = this.getChainId(networkClientId);
+    const smartTransactionsForChainId = smartTransactions?.[chainId];
 
-    const transactionsToUpdate: string[] = currentSmartTransactions
+    const transactionsToUpdate: string[] = smartTransactionsForChainId
       .filter(isSmartTransactionPending)
       .map((smartTransaction) => smartTransaction.uuid);
 
     if (transactionsToUpdate.length > 0) {
-      this.fetchSmartTransactionsStatus(transactionsToUpdate);
+      this.fetchSmartTransactionsStatus(transactionsToUpdate, {
+        networkClientId,
+      });
     }
   }
 
-  async confirmSmartTransaction(smartTransaction: SmartTransaction) {
+  // TODO make this a private method?
+  async confirmSmartTransaction(
+    smartTransaction: SmartTransaction,
+    {
+      chainId,
+      ethersProvider,
+    }: {
+      chainId: Hex;
+      ethersProvider: any;
+    },
+  ) {
     const txHash = smartTransaction.statusMetadata?.minedHash;
     try {
-      const transactionReceipt =
-        await this.ethersProvider.getTransactionReceipt(txHash);
-      const transaction = await this.ethersProvider.getTransaction(txHash);
+      const transactionReceipt = await ethersProvider.getTransactionReceipt(
+        txHash,
+      );
+      const transaction = await ethersProvider.getTransaction(txHash);
       const maxFeePerGas = transaction.maxFeePerGas?.toHexString();
       const maxPriorityFeePerGas =
         transaction.maxPriorityFeePerGas?.toHexString();
       if (transactionReceipt?.blockNumber) {
-        const blockData = await this.ethersProvider.getBlock(
+        const blockData = await ethersProvider.getBlock(
```

Comment:
Hmm... this isn't going to work anymore. We can't guarantee that `ethersProvider` is an Ethers provider; it could be our provider from NetworkController, in which case methods like this aren't supported.

Do we even have to use Web3Provider in this file? We have typically use `@metamask/eth-query`, which just uses `sendAsync` to make requests.

Reply:
Yeah I could sub out `Web3Provider` for `@metamask/eth-query` throughout this controller if that is the direction we've been heading?

Reply:
done here: https://github.com/MetaMask/smart-transactions-controller/commit/fe9732ccf000b24a670fca8681edc998f45771c2